src/platform/Tizen/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -0,0 +1,176 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <lib/support/CodeUtils.h>
+#include <lib/support/SafeInt.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <platform/Tizen/NetworkCommissioningDriver.h>
+
+#include <limits>
+#include <string>
+#include <vector>
+
+using namespace chip;
+using namespace chip::Thread;
+using namespace chip::DeviceLayer::Internal;
+
+namespace chip {
+namespace DeviceLayer {
+namespace NetworkCommissioning {
+
+#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
+namespace {
+constexpr char kWiFiSSIDKeyName[]        = "wifi-ssid";
+constexpr char kWiFiCredentialsKeyName[] = "wifi-pass";
+} // namespace
+
+CHIP_ERROR TizenWiFiDriver::Init()
+{
+    CHIP_ERROR err;
+    size_t ssidLen        = 0;
+    size_t credentialsLen = 0;
+
+    err = PersistedStorage::KeyValueStoreMgr().Get(kWiFiCredentialsKeyName, mSavedNetwork.credentials,
+                                                   sizeof(mSavedNetwork.credentials), &credentialsLen);
+    if (err == CHIP_ERROR_NOT_FOUND)
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    err = PersistedStorage::KeyValueStoreMgr().Get(kWiFiSSIDKeyName, mSavedNetwork.ssid, sizeof(mSavedNetwork.ssid), &ssidLen);
+    if (err == CHIP_ERROR_NOT_FOUND)
+    {
+        return CHIP_NO_ERROR;
+    }
+    mSavedNetwork.credentialsLen = credentialsLen;
+    mSavedNetwork.ssidLen        = ssidLen;
+
+    mStagingNetwork = mSavedNetwork;
+    return err;
+}
+
+CHIP_ERROR TizenWiFiDriver::CommitConfiguration()
+{
+    ReturnErrorOnFailure(PersistedStorage::KeyValueStoreMgr().Put(kWiFiSSIDKeyName, mStagingNetwork.ssid, mStagingNetwork.ssidLen));
+    ReturnErrorOnFailure(PersistedStorage::KeyValueStoreMgr().Put(kWiFiCredentialsKeyName, mStagingNetwork.credentials,
+                                                                  mStagingNetwork.credentialsLen));
+    mSavedNetwork = mStagingNetwork;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR TizenWiFiDriver::RevertConfiguration()
+{
+    mStagingNetwork = mSavedNetwork;
+    return CHIP_NO_ERROR;
+}
+
+bool TizenWiFiDriver::NetworkMatch(const WiFiNetwork & network, ByteSpan networkId)
+{
+    if (networkId.size() != network.ssidLen)
+    {
+        ChipLogProgress(NetworkProvisioning, "ssidLen is mismatched. network.ssidLen: %u, networkId.size(): %u", network.ssidLen,
+                        networkId.size());
+        return false;
+    }
+    else if (memcmp(networkId.data(), network.ssid, network.ssidLen) != 0)
+    {
+        ChipLogProgress(NetworkProvisioning, "ssid is mismatched. network.ssid: %s, networkId.data(): %s", network.ssid,
+                        networkId.data());
```

Comment:
networkId.data() does not have to be null-terminaged (and generally would not be).  How is that suposed to work?

Reply:
Is wifi ssid guaranteed to be null-terminated?
ssid is a maximum of 32 bytes null terminated string in specification.
but in case of Thread network, as you said, we need to consider not-null-terminated.

Reply:
>  ssid is a maximum of 32 bytes null terminated string in specification.

Which specification?

Reply:
I just checked wifi ssid specification.
Please tell me if the specification is different in matter wifi ssid.
actually, when I check the Linux platform code, wifi ssid and credential seem null-terminated string like below.
 932     memcpy(ssidStr, ssid.data(), ssid.size());
 933     memcpy(keyStr, credentials.data(), credentials.size());
 934     g_variant_builder_add(&builder, "{sv}", "ssid", g_variant_new_string(ssidStr));
 935     g_variant_builder_add(&builder, "{sv}", "psk", g_variant_new_string(keyStr));

Reply:
>  932 memcpy(ssidStr, ssid.data(), ssid.size());

That is definitely not null-terminated in general in Matter.

Any time you see a span, you can't assume null-termination (that's why it's carrying a size along).  In particular, spans that come from TLV decoding are not going to be null-terminated.

>  934 g_variant_builder_add(&builder, "{sv}", "ssid", g_variant_new_string(ssidStr));

And that will, still in general, not work right.  It's presumably working because `ssid.size() < kMaxWiFiSSIDLength`, so the trailing already-zeroed bytes are treated as null.  Filed https://github.com/project-chip/connectedhomeip/issues/15847

Reply:
Ok. I will modify it. Thank you for kindly review.